### PR TITLE
[config] android plugins

### PR DESCRIPTION
**Pull request by EvanBacon**
Wrap all existing config methods in plugins and unify the file writing for common operations. A lot of complex functionality is wrapped in dangerous mods pending more investigation on how to make them safer. mods like "android.strings" currently don't account for any kind of localization but this could possibly be provided to the mod as a `modRequest` property.

# Test Plan 

- Extended the iOS tests to account for Android.
  - added checks to ensure files are written in a valid format.

### CODE REVIEW

**Review by brentvatne — APPROVED** [review]

**Comment by brentvatne** [comment]
i took a stab at clarifying wording in a few places in 44a9391b85dc94920f053bc7db591f3087a09309 - feel free to revert / correct anything that is factually incorrect or that you think i made worse

File: packages/config/src/plugins/__tests__/expo-plugins-test.ts [hunk 1]
```diff
@@ -195,14 +212,53 @@ describe(withExpoIOSPlugins, () => {
         usesAppleSignIn: true,
         accessesContactNotes: true,
       },
-
+      android: {
+        package: 'com.bacon.tester.expoapp',
+        versionCode: 6,
+        backgroundColor: '#ff0000',
+        userInterfaceStyle: 'light',
+        // icon: './icons/android-icon.png',
```
Comment (EvanBacon) [inline packages/config/src/plugins/__tests__/expo-plugins-test.ts:220]:
```suggestion
```

File: packages/config/src/plugins/__tests__/expo-plugins-test.ts [hunk 2]
```diff
@@ -164,7 +181,7 @@ describe(withExpoIOSPlugins, () => {
       facebookScheme: 'fb1234567890',
       ios: {
         // publishManifestPath: './ios-manifest'
-        publishBundlePath: './ios-dist',
+        // publishBundlePath: './ios-dist',
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply packages/config/src/plugins/__tests__/expo-plugins-test.ts:184]:
```suggestion
```

File: packages/config/src/plugins/expo-plugins.ts [hunk 3]
```diff
@@ -49,8 +50,59 @@ export const withExpoIOSPlugins: ConfigPlugin<{
...
+    AndroidConfig.NavigationBar.withNavigationBar,
+    AndroidConfig.StatusBar.withStatusBar,
+    AndroidConfig.PrimaryColor.withPrimaryColor,
+
+    AndroidConfig.Icon.withIcons,
+    AndroidConfig.SplashScreen.withSplashScreen,
+
+    // If we renamed the package, we should also move it around and rename it in source files
+    // Added last to ensure this plugin runs first. Out of tree solutions will mistakenly resolve the package incorrectly otherwise.
+    AndroidConfig.Package.withPackageRefactor,
```
Comment (bbarthec) [inline packages/config/src/plugins/expo-plugins.ts:106]: This looks awesome 🤩

**Review by bbarthec — APPROVED** [review]
I really like the direction it is going! 👍 
So, my understanding is that we have built-in suport for modified limited set of files (`AndroidManifest`, two `.gradle` files and a bunch of files in `res` directory), but we also can modify arbitrary `xml/java/kt` file by using some `dangerous` technique? 🤔

**Pull request merged**